boilerplates/aws-lambda-serverless/files/serverless.yml

```diff
@@ -0,0 +1,243 @@
+service: ${self:custom.project}-${self:provider.stage}
+frameworkVersion: '3'
+configValidationMode: error
+
+provider:
+  name: aws
+  runtime: nodejs20.x
+  architecture: arm64
+  deploymentMethod: direct
+  region: us-east-1 # North Virginia
+  stage: ${opt:stage, 'dev'}
+  versionFunctions: false # true is required for lamda edge deployment
+
+plugins:
+  - serverless-esbuild
+  - serverless-s3-sync
+  - serverless-cloudfront-invalidate
+  - serverless-scriptable-plugin
+
+functions:
+  vikeapp:
+    handler: hono-entry_aws_lambda.handler
+    environment:
+      NODE_ENV: production
+    events:
+      - httpApi: "*"
+
+custom:
+  project: vike
+  scriptable:
+    # add custom hooks
+    hooks:
+      before:package:createDeploymentArtifacts: 
+        - pnpm build
+  esbuild:
+    config: './esbuild.config.cjs'
+
+  s3Sync:
+    - bucketNameKey: WebsiteStaticBucketnameKey
+      localDir: dist/client
+      deleteRemoved: true
+      acl: public-read # required
+      params:
+        - "assets/pwa/sw.js":
+            CacheControl: 'no-cache'
+            #ServiceWorkerAllowed: '/'
+        - "assets/pwa/manifest.webmanifest":
+            CacheControl: 'no-cache'
+        - "*.*":
+            CacheControl: 'public, max-age=31536000'
+  cloudfrontInvalidate:
+    - distributionIdKey: 'FrontendCloudFrontDistributionId'
+      autoInvalidate: true
+      items:
+        - '/*'
```

Comment:
I'm not familiar with Serverless. How is this block used, and by what?
Is this necessary to have `./esbuild.config.cjs` as a CJS module? Can't it be MJS?

Reply:
I have looked into converting `esbuild.config.cjs` into mjs or ts but the serverless plugin [serverless-esbuild](https://github.com/floydspace/serverless-esbuild) fails executing it in any other format than commonjs.